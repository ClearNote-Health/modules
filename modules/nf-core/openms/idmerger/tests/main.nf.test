nextflow_process {

    name "Test Process OPENMS_IDMERGER"
    script "../main.nf"
    process "OPENMS_IDMERGER"

    tag "modules"
    tag "modules_nfcore"
    tag "openms"
    tag "openms/idmerger"

    test("proteomics - openms - idxml") {

        when {
            process {
                """
                input[0] = Channel.of([
                            [ id:'test'], // meta map
                            [ file(params.test_data['proteomics']['openms']['idxml1'], checkIfExists: true),
                              file(params.test_data['proteomics']['openms']['idxml2'], checkIfExists: true)]
                        ])
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.versions).match("versions") },
                { assert snapshot(process.out.idxml).match("idxml") }
            )
        }

    }

    test("proteomics - openms - idxml - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = Channel.of([
                            [ id:'test'], // meta map
                            [ file(params.test_data['proteomics']['openms']['idxml1'], checkIfExists: true),
                              file(params.test_data['proteomics']['openms']['idxml2'], checkIfExists: true)]
                        ])
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
