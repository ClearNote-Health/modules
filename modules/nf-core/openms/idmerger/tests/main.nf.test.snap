{
    "versions": {
        "content": [
            [
                "versions.yml:md5,dc4fa28b55f18860166ada8bae8faaaa"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-19T14:12:47.962180653"
    },
    "idxml": {
        "content": [
            [
                [
                    {
                        "id": "test"
                    },
                    "test.idXML:md5,4bbb5407419f2829b08bc20fe6bd3a64"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-19T14:12:47.974368373"
    },
    "proteomics - openms - idxml - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.idXML:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,dc4fa28b55f18860166ada8bae8faaaa"
                ],
                "idxml": [
                    [
                        {
                            "id": "test"
                        },
                        "test.idXML:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,dc4fa28b55f18860166ada8bae8faaaa"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-19T14:12:56.772745939"
    }
}